### DISCUSSION TITLE
core:common 모듈

### DISCUSSION THREAD

**Discussion by chanubc** [Team-Wable/WABLE-ANDROID#12] [Ideas]
## 👋  feature:main, feature:main-compose 모듈간 공통되는 사항들 common 모듈이 필요할까요?
![image](https://github.com/user-attachments/assets/09e6850a-f034-47f2-bb2b-ab3610e383ee)

현재 우리의 그래프는 위와 같습니다. 아주 난리가 났는데요.
컴포즈 관련한 의존성들을 모두 auth에 놓아도 되었겠지만 추후에 compose로 마이그레이션 하는 기능이 존재 할 수 있어 최대한 모듈 분리를 해 놓았습니다.

### 모듈 설명
1. main-compose : entry를 담당, view container, nav host, navigation, -> xml로 치면 MainAcitivty와, navController, mainViewConatainer, nav_graph.xml 이 존재한다 생각하시면 됩니다.
2. navigation : 경로를 담당하는 data class, 모든 모듈에서 참조해야 하기에 분리가 필요함.
3. designsystem-compsoe : 컴포즈 theme, type, color를 담당, component도 담당(버튼 등등)

### 문제점 
1. 모든 compose feature 모듈은 designsystem을 의존해야함 -> icon들이 여기 들어가 있기 때문. icon 때문에 불필요한 xml파일들에 접근 가능함.
2. main과 main-compose 모듈간 intent로 이동해야하는데, 이러면 common 모듈에 intent를 가능케 하는 interface가 필요합니다.
3. 모든 컴포즈 feature 모듈도 ui를 의존해야합니다. -> 앰플리튜드

또 ui모듈에 앰플리튜드가 들어가기에 모든 모듈이 ui모듈을 참조해야 합니다. ui모듈의 경우 xml기반의 코드들이 존재하는데 앰플리튜드의 존재로 인해 ui모듈 또한 의존해야 합니다.
현재는 의존성이 지나치다 생각합니다.

### 결론
1. common모듈을 만들고 common모듈에 icon, 앰플리튜드, intent provider(interface)를 만드는 방안
기존의 designsystem모듈에는 xml관련 themes, font, color, string등등을 넣으면 될것 같습니다!
2. 이미 designsytem은 완료가 되었으니 common 모듈에 앰플리튜드, intent provider(interface)를 넣는 방안
3. 그냥 designsytem에 icon,  앰플리튜드, intent provider(interface)를 때려 박는 방안

3가지 방안중에 괜찮은 방안이 있을까요?

좋은 의견주시면 제공해주세요! 수업중이라 두서 없이 썼네요ㅠ
@sohyun127 @Eonji-sw

**Comment by sohyun127** [comment]
1번안이 제일 나은거 같아요! 근데 그렇게 했을 때 남는 기존 designsystem 모듈 기능이 많이 없는데 과하게 모듈 분리를 한거 같아서 차라리 남는 font..등등 xml용에서 쓰이는거는 core:ui에서 관리하는게 어떤가요? 컴포즈와 xml에서 같이 쓰이는 것들은 1번안처럼 common에서 관리하구요!

**Reply by chanubc** [reply]
**designsystem -> common** (rename) : icon, 앰플리튜드, intent provider(interface), font.ttf(pretandard파일)
**ui** -> xml용(font, color), shape, string등등 

(그런데 string.xml만 놓고 보면 재사용 되는 부분이 많이 없는 것 같아서 재사용하지 않는 string들은 각 feature 모듈에서 사용하는건 어떨까요?)
~~(비슷한 예로 icon들도 share네이밍이 붙는게 아니라면 각 feature에 넣어도 될것 같아요!)~~

**Reply by sohyun127** [reply]
음 재사용과 아닌 것들을 분리하면 그게 맞긴한데, 혼재되어 있어서 나중에 한번에 보기가 좀 힘들진 않을지 걱정되긴 하네요!! 개인적을 한곳에 있는게 마음 편안한 느낌..?ㅎㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ string, dimen 파일의 경우는 컴포즈 뷰에서도 쓰이는 게 있을테니 common에 그냥 둬도 될 것 같다는 의견입니다! 고민되네요 의견이 궁금합니다ㅎㅎ

**Reply by Eonji-sw** [reply]
저도 1안이 가장 좋다고 생각해요. 공통 부분만 common에서 처리하는 것이 훨 좋아보이고 역할도 명확할 것 같네요!

string.xml은 분리하면 너무 번거롭지 않을까.. 굳이라는 생각은 드네용.. 하지만 저희 플젝 GUI를 보면 재사용되는 글자가 딱히 없는 것 같네요! 그러나 추후 겹치는 글자가 생길 경우 각 모듈 string에 방문하여 이를 확인해야 한다면 별로인 것 같네요..

**Reply by chanubc** [reply]
1안으로 수정하겠습니다1
